package/system/apk/patches/0002-add-no-chroot-option.patch

```diff
@@ -0,0 +1,47 @@
+From 239efe9892a9bfecaadff10c37095eff81cd33cd Mon Sep 17 00:00:00 2001
+From: Paul Spooren <mail@aparcar.org>
+Date: Thu, 9 Dec 2021 07:17:42 -1000
+Subject: [PATCH 2/5] add --no-chroot option
+
+Signed-off-by: Paul Spooren <mail@aparcar.org>
+---
+ doc/apk-add.8.scd | 3 +++
+ src/app_add.c     | 4 ++++
+ 2 files changed, 7 insertions(+)
+
+diff --git a/doc/apk-add.8.scd b/doc/apk-add.8.scd
+index 1586f7e..52b339e 100644
+--- a/doc/apk-add.8.scd
++++ b/doc/apk-add.8.scd
+@@ -53,3 +53,6 @@ following options:
+ 	ownership when running as root. However, this option is turned on when
+ 	running as non-root user, as changing file ownership is not permitted
+ 	by the system then.
++
++*--no-chroot*
++	Do not run chroot for scripts.
+diff --git a/src/app_add.c b/src/app_add.c
+index d12d2a3..e949065 100644
+--- a/src/app_add.c
++++ b/src/app_add.c
+@@ -28,6 +28,7 @@ struct add_ctx {
+ 	OPT(OPT_ADD_initdb,	"initdb") \
+ 	OPT(OPT_ADD_latest,	APK_OPT_SH("l") "latest") \
+ 	OPT(OPT_ADD_no_chown,	"no-chown") \
++	OPT(OPT_ADD_no_chroot,	"no-chroot") \
+ 	OPT(OPT_ADD_upgrade,	APK_OPT_SH("u") "upgrade") \
+ 	OPT(OPT_ADD_virtual,	APK_OPT_ARG APK_OPT_SH("t") "virtual")
+ 
+@@ -47,6 +48,9 @@ static int option_parse_applet(void *ctx, struct apk_ctx *ac, int opt, const cha
+ 	case OPT_ADD_no_chown:
+ 		actx->extract_flags |= APK_FSEXTRACTF_NO_CHOWN;
+ 		break;
++	case OPT_ADD_no_chroot:
++		ac->flags |= APK_NO_CHROOT;
```

Comment:
This should probably be a global flag instead of just on `apk add`, because you can do other operations like `apk fix`.